modules/geolocation/geolocationWorker.py

```diff
@@ -0,0 +1,92 @@
+from modules.geolocation.geolocation import Geolocation
+import multiprocessing as mp
+# For exception handling
+import queue
+
+
+def exit_requested(requestQueue):
+    return not requestQueue.empty()
+
+
+def geolocation_locator_worker(pause, exitRequest, pipelineIn, pipelineOut, pipelineOutLock):
+
+    print("Start Geolocation")
+    locator = Geolocation()
+
+    while True:
+
+        pause.acquire()
+        pause.release()
+
+        # Pixel coordinates of tents and plane data
+        coordinates = pipelineIn.get()
+        # Check for valid input
+        if (coordinates is None):
+            continue
+
+        ret, location = locator.run_locator(coordinates)
+
+        # Something has gone wrong, skip
+        if (not ret):
+            continue
+
+        # External lock required as geolocation_output_worker will dump the queue (without using the internal one)
+        pipelineOutLock.acquire()
+        pipelineOut.put(location)
```

Comment:
I think there's a deadlock condition here? The [put method blocks](https://docs.python.org/3/library/multiprocessing.html#multiprocessing.Queue.put) if the pipeline is full (don't think we're determining the size of the queue right now but there should be some kind of innate memory limit, as described in [this issue](https://bugs.python.org/issue8237). Then, this process has the piplineLock, so the output_worker cannot take from the location to empty the pipeline. At the same time, the put method cannot put anything into the pipeline, so it won't release the lock.

Not sure though, not that comfortable with concurrency theory yet. If we do want a work around, checkout [put_nowait ](https://docs.python.org/3/library/multiprocessing.html#multiprocessing.Queue.put_nowait), if the queue is full just have it release the lock and continue, maybe?